### ISSUE TITLE
"no-return-await" fails inside try-catch

### ISSUE THREAD

**Issue by vitaly-t** [palantir/tslint#4824]
### Bug Report

- __TSLint version__: 5.18.0
- __TypeScript version__: 3.5.3
- __Running TSLint via__: Node.js

#### Reproduction using [TSLint Playground](https://palantir.github.io/tslint-playground)

[TSLint Playground Link](https://palantir.github.io/tslint-playground/?saved=N4Igxg9gJgpiBcICGBnAngOzAAgGYFcsAXASwg2yJhSIB4AVAPgAoBKebABQCcIBbEihgNG2YAB0K2aZAw1sAB2wBebBhgB3LrwFDm3ahAA2ANxgrRQovRJ8YEfEX2HTMVqwDckydOlFuaGI+viEGRPjcFEgaSCREigB0RAAWMBgibB7YAPTZ2ACiAEqFAPKFHCkkGADWKNjRsfGCajAwsFDYqQbB0gC+2GBIRGDJzG5BUn3ek9hhERQKSanpTJk5eZDcBmBERoGVNXUpQ-UxcdjNGBDx6m1tnTDdGL3TIAA04OS4JADmCCC5bAATQcAyQFD40BIuH2qQGX1+ESGZAoXXMGjiyWwaAc3GwEA0GASkkBAFkkNVzCgIuZjvEcRFsAArFDkC51ExIIwkKBveoYDrQ7Gg7iEC4YIgQEl5a5ddnU6h8-yBEYwMDVKo-Shw2SsozmXAQPGPXjcFDEjCSCQzcQgUX6lC2jjWkIhW1XAC0c0iHoacSdlFFMB62H6gKQUA6EagcRRXNm+Ad-I6KRgJDxsm+PyRpHIkheGBAvSAA).

#### TypeScript code being linted

```ts
async function test<T>(): Promise<T> {
    const p = new Promise(resolve => setTimeout(resolve));

    try {
        return await p.then<T>(); // PROBLEM: thinks await is needed here
    } catch(e){

    }

    return p.then<T>(); // correctly thinks that await is not needed here
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-return-await": true
    }
}
```

#### Actual behavior

Suggests that `await` is needed after `return` and before `p.then<T>()` when inside `try->catch` block.

#### Expected behavior

Error on using `await` after `return`.

**Comment by fatcerberus** [comment]
I think that’s right: if you just `return promise;` inside a `try` and it rejects, then the code in the `catch` block won’t run.  IIRC this is the only case where `return await` differs from `return`.

**Comment by vitaly-t** [comment]
Wow, thanks! This is can be difficult to wrap brains around at first.
